test/firebase_auth_mocks_test.dart

```diff
@@ -390,6 +390,25 @@ void main() {
     expect(user.displayName, 'New Bob');
   });
 
+  test('User.updateEmail works', () async {
+    final auth = MockFirebaseAuth(signedIn: true, mockUser: tUser);
+    final user = auth.currentUser;
+    expect(
+      () async => await user!.updateEmail('john@domain.tld'),
+      returnsNormally,
+    );
```

Comment:
Can you add one additional check to verify that the email was updated to the new value?

Also, this test actually changes the value of tUser, which can potentially affect other tests. Would you mind either: 1) initializing your own instance of MockUser in this test
or
2) adding a `setUp` at the top which reinitializes tUser at every test? Something like:
```dart
late MockUser tUser;
setUp(()  {
  tUser = MockUser(
    isAnonymous: false,
    //some other values
  );
});
```

1) might be easier for you to do, while 2) is actually the better way, but might require more work. If you do 1), I'll file a ticket to do 2) in a separate PR.